### ISSUE TITLE
Messages from a pull consumer are not reliably received by `consumer.next()` when nats connection is slow.

### ISSUE THREAD

**Issue by awesomestmason** [nats-io/nats.deno#731]
### Observed behavior

Messages from a pull consumer are not reliably received by `consumer.next()` when nats connection is slow. I have experienced this after deploying a service using the nats library, and observed messages taking 3-4 redeliveries until they are received by my service. In my setup:
1. I have cross-account jetstream, this could be adding some latency
2. I set the "pull interval"/"expiration" to 1s, instead of 30s (which particularly highlights this issue, because the current padding is +5% => 50ms). 
`consumer.next({ expires: 1000 });`

I have narrowed it down to be caused by this line:
https://github.com/nats-io/nats.deno/blob/7ba5c6e3d4c87bd35665d1a0ff4156a21070dcfc/jetstream/consumer.ts#L890
Printing out the internal jetstream messages received by my consumer - I observed NO expected "408 request timeout" responses from jetstream. And it also tends to miss the actual message deliveries, depending on how far into the timeout they are delivered.

Switching the padding to be higher, for example `* 2.5` I observed that I began receiving the "408 request timeout" responses, and all messages were delivered.

### Expected behavior

I would request that either the timeout padding is configurable, or increased. The C# client for example, adds a constant padding of 5s, instead of 5 percent. (and for reference I did not have this same issue with the C# client).
https://github.com/nats-io/nats.net/blob/dcf77a41b0fffe522ff6b6afcd5d8631cb7e1cba/src/NATS.Client.JetStream/Internal/NatsJSFetch.cs#L123C1-L124C1

### Server and client version

server version: 2.1.0.18
client version: 1.28.2

### Host environment

Running in kubernetes, with multi-account NATS setup

### Steps to reproduce

Reproduction is difficult here because I'm not sure the best way to purposefully introduce latency to NATS.

**Comment by aricart** [comment]
@awesomestmason, the main problem is that your expires are sensitive to the cluster/connection—very short expires slam the server in multiple ways, further aggravating the situation. The extra padding, as you suggest, would be a bad idea because a recommended expiration of 30s (the default) would result in a 1m timeout, which doesn't line up with the expectation.

Short timeouts like you are doing are generally a bad idea because they create a lot of stress on the server. If there are no messages, you'll continuously re-poll. In cases where the client gives up because the timeout is too aggressive, the work performed by the server is wasted (creating additional load).

**Comment by awesomestmason** [comment]
@aricart Thanks for taking a look at this issue. 

I would like to clarify that I initially had the default expiration for my consumer, but I noticed some messages were failing to be delivered (if they were close to the end of the expiration). It was perhaps 10% of messages, but enough to start an investigation. I then noticed as I decreased the expiration timer, a higher percentage of message deliveries were missed.

So the issue was actually first discovered with the default 30s expiration (1.5s padding).

> The extra padding, as you suggest, would be a bad idea because a recommended expiration of 30s (the default) would result in a 1m timeout, which doesn't line up with the expectation.

I agree a 1min timeout seems too long, I mentioned the `2.5x` figure as an extra-wide padding window that helped me verify that my issue was indeed caused by the timeout padding. I think realistically, a constant 5s padding, like the C# client, would make sense.

And I would also push on why should this padding be a percentage? Is the thinking that the larger the expiration, the more likely for the clocks to get out of sync? I would think the padding time should be some constant padding + some percentage. Where the constant padding would account for network latency, and the percentage would account for clock-desync.
`padding = (5000) + (0.05 * expires)`
